src/Presentation/Presenters/DonationFormViolationPresenter.php

```diff
@@ -114,4 +116,36 @@ private function getBankData( AddDonationRequest $request ): array {
 		];
 	}
 
+	private function getValidationResult( array $violations ): array {
+		$fieldGroups = [
+			Result::SOURCE_PAYMENT_AMOUNT => 'paymentData',
+			Result::SOURCE_PAYMENT_TYPE => 'paymentData',
+
+			Result::SOURCE_BANK_ACCOUNT => 'bankData',
+			Result::SOURCE_BANK_CODE => 'bankData',
+			Result::SOURCE_IBAN => 'bankData',
+			Result::SOURCE_BIC => 'bankData',
+
+			Result::SOURCE_DONOR_EMAIL => 'address',
+			Result::SOURCE_DONOR_COMPANY => 'address',
+			Result::SOURCE_DONOR_FIRST_NAME => 'address',
+			Result::SOURCE_DONOR_LAST_NAME => 'address',
+			Result::SOURCE_DONOR_SALUTATION => 'address',
+			Result::SOURCE_DONOR_TITLE => 'address',
+			Result::SOURCE_DONOR_STREET_ADDRESS => 'address',
+			Result::SOURCE_DONOR_POSTAL_CODE => 'address',
+			Result::SOURCE_DONOR_CITY => 'address',
+			Result::SOURCE_DONOR_COUNTRY => 'address',
+
+		];
+		return array_reduce( $violations, function( $validationResult, ConstraintViolation $violation ) use ( $fieldGroups ) {
```

Comment:
When code is too long for a single line like this I really appreciate having one argument per line, like this:

```php
array_reduce(
    $input,
    function( $arguments ) use ( $stuff ) {
        doThings();
    } 
);
```